sys/sysent.c

```diff
@@ -24,8 +27,58 @@ int sys_write(thread_t *td, syscall_args_t *args) {
   return -EBADF;
 }
 
+/* This is just a stub. A full implementation of this syscall will probably
+   deserve a separate file. */
+/* Note that this sbrk implementation does not actually extend .data section,
+   because we have no guarantee that there is any free space after .data in the
+   memory map. But it does not matter much, because no application would assume
+   that we are actually expanding .data, it will use the pointer returned by
+   sbrk. */
+int sys_sbrk(thread_t *td, syscall_args_t *args) {
+  size_t increment = (size_t)args->args[0];
+
+  kprintf("[syscall] sbrk(%zu)\n", increment);
+
+  if (!td->td_uspace) {
+    log("sbrk called by a thread with no userspace map");
+    return -ENOMEM;
+  }
+
+  if (!td->td_brk_entry) {
+    /* sbrk was not used before by this thread. */
+    size_t size = roundup(increment, PAGESIZE);
+    vm_addr_t addr;
+    if (vm_map_findspace(td->td_uspace, 0x08000000, size, &addr) != 0)
```

Comment:
Magic constant alert! Please name it and provide a comment why you chose this specific value.